lib/screens/route_patterns/route_direction_stops.ex

```diff
@@ -0,0 +1,82 @@
+defmodule Screens.RoutePatterns.RouteDirectionStops do
+  @moduledoc false
+
+  require Logger
+
+  def parse_result(%{"data" => data, "included" => included}, route_id) do
+    included_data = parse_included_data(included)
+    parse_data(data, included_data, route_id)
+  end
+
+  def parse_result(_, _) do
+    Logger.warning("Unrecognized format of route_pattern data.")
+    :error
+  end
+
+  defp parse_included_data(data) do
+    data
+    |> Enum.map(fn item ->
+      {{Map.get(item, "type"), Map.get(item, "id")}, parse_included(item)}
+    end)
+    |> Enum.into(%{})
+  end
+
+  defp parse_included(%{"type" => "stop"} = item) do
+    Screens.Stops.Parser.parse_stop(item)
+  end
+
+  defp parse_included(%{
+         "type" => "trip",
+         "relationships" => %{"stops" => %{"data" => stops_data}}
+       }) do
+    Enum.map(stops_data, &parse_stop_data/1)
+  end
+
+  defp parse_stop_data(%{"id" => stop_id, "type" => "stop"}) do
+    stop_id
+  end
+
+  defp parse_data(data, included_data, route_id) do
+    filtered_data = filter_by_route(data, route_id)
+    [typical_data | _] = filtered_data
```

Comment:
Can this just be 
```suggestion
    [typical_data | _] = filter_by_route(data, route_id)
```

Reply:
Probably! I didn't write this at all — this is a rename of the original `Parser` that's not being detected as a rename, I assume because the "from" file still exists and has new contents. I think by splitting this up into a commit that does the rename, and a second commit that introduces the new `Parser`, it will be detected properly (and then blames and stuff will still work).

Reply:
...also these changes are incoherent, since I accidentally split the rename across 8f32b42 and 89cdbc0 🫠